bindings/java/src/convert.rs

```diff
@@ -0,0 +1,75 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+use jni::errors::Result;
+use jni::objects::{JMap, JObject, JString, JThrowable, JValue};
+use jni::JNIEnv;
+use opendal::ErrorKind;
+use std::collections::HashMap;
+
+pub(crate) fn error_to_error<E>(error: E) -> opendal::Error
+where
+    E: Into<anyhow::Error> + ToString,
+{
+    opendal::Error::new(ErrorKind::Unexpected, &error.to_string()).set_source(error)
+}
+
+pub(crate) fn error_to_exception<'local>(
+    env: &mut JNIEnv<'local>,
+    error: opendal::Error,
+) -> Result<JThrowable<'local>> {
+    let class = env.find_class("org/apache/opendal/exception/ODException")?;
+
+    let code = env.new_string(match error.kind() {
```

Comment:
Maybe we can use `error.kind().to_string()`?

Reply:
It comes back to the previous pull request discussion that we may cause a runtime exception on unknown error kind - here we set the default error kind as Unexpected.